Cancelling staked expenditure

jakubcolony opened the pull request:
## Description

A staked expenditure can be cancelled by anyone with arbitration permissions or using motions (#1061) and a choice can be made whether to punish the creator by forfeiting their stake.

## Testing
1. Enable reputation monitor. Pay yourself some tokens and activate them.
2. Install and enable Staked Expenditure extension.
3. Create an expenditure by staking it and lock it.
4. Cancel staked expenditure button should appear. In the dialog that it opens you should be able to select whether to punish the creator.
5. `cancelStakedExpenditure` saga should fire. After page refresh, the expenditure status should change to Cancelled. 
6. If you chose not to punish the creator, you should be able to reclaim the stake.

Resolves #1033

willm30 reviewed (commented):
Nice, looking good. 

Couple of notes:

1. Should cancellation still be possible after finalization and claim?
![cancel](https://github.com/JoinColony/colonyCDapp/assets/64402732/0e179f39-6564-48a3-be2d-86f2880a6cfc)

2. I'm not getting any UI feedback after a few refreshes when cancelling the other way, i.e. before an expenditure is locked. It does go through on chain, so when you click it again or click lock you get an error.
![cancel2](https://github.com/JoinColony/colonyCDapp/assets/64402732/3f46d03a-b413-4384-80fe-9cfd5a87b31b)

jakubcolony commented:
@willm30 

> Should cancellation still be possible after finalization and claim?

Not after claiming I suppose, but I'll need to check with Arren on cancelling after finalization before claiming. For now, I changed it to only allow cancelling while in locked state.

> I'm not getting any UI feedback after a few refreshes when cancelling the other way, i.e. before an expenditure is locked. It does go through on chain, so when you click it again or click lock you get an error.

That was a contract issue that's been now fixed, the PR's been updated to run the newest network changes.

willm30 reviewed (approved):
LGTM
